### PR TITLE
Resuming for `InfiniteSampler`

### PR DESCRIPTION
**Pull request by juliusberner**
# PhysicsNeMo Pull Request

## Description
When resuming before iterating through the whole dataset, the `InfiniteSampler` would restart and a certain part of the dataset would be ignored during training. We also adapt the seed based on `cur_nimg` to avoid the same randomness in, e.g., dropout, after resuming.

## Checklist

- [X] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.
- [X] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

### CODE REVIEW

File: examples/weather/corrdiff/train.py [hunk 1]
```diff
@@ -153,8 +153,17 @@ def main(cfg: DictConfig) -> None:
             cfg.training.hp.total_batch_size // dist.world_size
         )
 
+    # Load the current number of images for resuming
+    try:
+        cur_nimg = load_checkpoint(
+            path=checkpoint_dir,
+            device=dist.device,
```
Comment (CharlelieLrt) [inline examples/weather/corrdiff/train.py]: We probably don't need the `device` argument since we are only reading `cur_nimg`

File: examples/weather/corrdiff/datasets/dataset.py [hunk 2]
```diff
@@ -110,14 +111,19 @@ def init_train_valid_datasets_from_config(
...
     """
 
     config = copy.deepcopy(dataset_cfg)
     (dataset, dataset_iter) = init_dataset_from_config(
-        config, dataloader_cfg, batch_size=batch_size, seed=seed
+        config,
+        dataloader_cfg,
+        batch_size=batch_size,
+        seed=seed,
+        sampler_start_idx=sampler_start_idx,
```
Comment (CharlelieLrt) [inline examples/weather/corrdiff/datasets/dataset.py:126]: Why is this not applied to the `validation_dataset` just below?

**Review by CharlelieLrt — CHANGES_REQUESTED** [review]
Just two minor questions/comments, otherwise everything looks good

File: examples/weather/corrdiff/datasets/dataset.py (hunk 2 above)
Comment (juliusberner) [reply examples/weather/corrdiff/datasets/dataset.py:126]: I could also include resuming for the validation sampler. However,
1) we currently do not count the nimg for the validation set 
2) we typically validate over the full validation dataset or do not care so much if we resume from the previous index. 

What do you think?

File: examples/weather/corrdiff/train.py (hunk 1 above)
Comment (juliusberner) [reply examples/weather/corrdiff/train.py]: I changed it, thanks!

File: examples/weather/corrdiff/datasets/dataset.py (hunk 2 above)
Comment (CharlelieLrt) [reply examples/weather/corrdiff/datasets/dataset.py:126]: Agree with 2. Thanks for the clarifications!

**Review by CharlelieLrt — APPROVED** [review]
LGTM, thanks for the fixes!
